agent/internal/config/secret.go

```diff
@@ -28,37 +29,43 @@ func (field *ConfigFromFile) SetValue(location string) error {
 	return nil
 }
 
-func checkGenerateKeys(location string) (string, error) {
-	log.Printf("Checking key file: %v\n", location)
-	file, err := os.ReadFile(location) //#nosec G304 -- location comes from an env
+func checkGenerateKeys(path string) (string, error) {
+	log.Printf("Checking key file: %v\n", path)
+	readFile, err := os.ReadFile(path) //#nosec G304 -- path comes from an env
+
+	if errors.Is(err, syscall.EISDIR) {
```

Comment:
Chaining the if-s (if-else if-else) would look better than the current if if-else. Opinions?

Reply:
Talked in person.